compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -578,10 +585,18 @@ private BType silentIntTypeCheck(BLangLiteral literalExpr, Object literalValue,
     private BType silentCompatibleLiteralTypeCheck(BFiniteType finiteType, BLangLiteral literalExpr,
                                                       Object literalValue, AnalyzerData data) {
         BType resIntType = symTable.semanticError;
+        List<BType> compatibleTypes = new ArrayList<>();
         for (BLangExpression valueExpr : finiteType.getValueSpace()) {
             resIntType = silentIntTypeCheck(literalExpr, literalValue, valueExpr.getBType(), data);
             if (resIntType != symTable.semanticError) {
-                return resIntType;
+                compatibleTypes.add(resIntType);
```

Comment:
Can we confirm that `resIntType` is not always a referenced type?

Reply:
`resIntType` is a type of a integer literal. So it can be either `byte`, `decimal`, `float` and `int`

Reply:
Shall we rename the variable? Kinda implies an int type.

Reply:
I think the name is ok. It is the type of the integer. wdyt?

Reply:
It is the type of the integer literal, not integer.

Reply:
Updated c9505a5002babe8ddffca64eb5b46dcb9a7eca05